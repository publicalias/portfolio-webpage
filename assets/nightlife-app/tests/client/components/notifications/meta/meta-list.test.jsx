"use strict";

//local imports

const MetaList = require("../../../../../scripts/client/components/notifications/meta/meta-list");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { initTestEvent, initTestSnapshot, withDataList } = require("redux/tests/client-tests");
const { testMock } = require("redux/tests/meta-tests");
const { reactTests } = require("redux/tests/react-tests");

//node modules

const React = require("react");

//utilities

const { testMount, testShallow } = testWrapper(MetaList);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(MetaList));

//meta list

describe("MetaList", () => {

  const dataList = [null, {
    bool: "[]",
    heading: "Meta",
    type: "meta"
  }, {
    children: []
  }];

  const testList = withDataList(testShallow, dataList);
  const testListMount = withDataList(testMount, dataList);

  const testSnapshot = initTestSnapshot(testList);

  const testClick = initTestEvent(testList, "click");

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (list)", () => testSnapshot(null, null, { children: [<div key="" />] }));

  it("should call refresh conditionally on update", () => {

    const refresh = jest.fn();

    const { setProps, wrapper } = testListMount(null, { refresh });

    wrapper.mount();

    setProps(null, null, { location: { pathname: "/users/list" } });

    testMock(refresh, [], []);

    wrapper.unmount();

  });

  it("should call handleClick on click", () => {

    const refresh = jest.fn();

    return testClick(".qa-refresh-list", [null, { refresh }], () => {
      testMock(refresh, []);
    });

  });

});
