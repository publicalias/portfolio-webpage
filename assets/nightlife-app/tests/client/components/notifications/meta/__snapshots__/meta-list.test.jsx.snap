// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MetaList should match snapshot (default) 1`] = `
<div
  className="c-notification-list--meta"
>
  <div
    className="c-notification-list__head"
  >
    <h3>
      Meta
    </h3>
    <button
      className="c-icon-button qa-refresh-list"
      onClick={[Function]}
    >
      <i
        className="fas fa-sync"
      />
    </button>
  </div>
  <hr />
  <div
    className="c-notification-list__body"
  >
    <p
      className="u-margin-none"
    >
      No Notifications
    </p>
  </div>
</div>
`;

exports[`MetaList should match snapshot (list) 1`] = `
<div
  className="c-notification-list--meta"
>
  <div
    className="c-notification-list__head"
  >
    <h3>
      Meta
    </h3>
    <button
      className="c-icon-button qa-refresh-list"
      onClick={[Function]}
    >
      <i
        className="fas fa-sync"
      />
    </button>
  </div>
  <hr />
  <div
    className="c-notification-list__body"
  >
    <div
      key=""
    />
  </div>
</div>
`;
